prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+        every {
+            mockFacade.putSetting(any(), any(), any(), Sender::class.java, any())
+        } returns Pair(SettingsFacade.AccessResult.SUCCESS, "")
+        every {
+            mockFacade.putSetting(any(), any(), any(), OrganizationAPI::class.java, any())
+        } returns Pair(SettingsFacade.AccessResult.SUCCESS, "")
+
+        return mockFacade
+    }
+
+    private fun setupSettingsFunctionForTesting(
+        oktaClaimsOrganizationName: String,
+        facade: SettingsFacade
+    ): SettingsFunction {
+        val claimsMap = buildClaimsMap(oktaClaimsOrganizationName)
+        val settings = MockSettings()
+        val sender = CovidSender(
+            name = "default",
+            organizationName = organizationName,
+            format = Sender.Format.CSV,
+            customerStatus = CustomerStatus.INACTIVE,
+            schemaName = "one"
+        )
+        settings.senderStore[sender.fullName] = sender
+        val sender2 = CovidSender(
+            name = "default",
+            organizationName = otherOrganizationName,
+            format = Sender.Format.CSV,
+            customerStatus = CustomerStatus.INACTIVE,
+            schemaName = "one"
+        )
+        settings.senderStore[sender2.fullName] = sender2
+
+        val receiver = Receiver(
+            "elr-secondary",
+            organizationName,
+            Topic.FULL_ELR,
+            CustomerStatus.TESTING,
+            "schema1"
+        )
+        settings.receiverStore[receiver.fullName] = receiver
+
+        val receiver2 = Receiver(
+            "test-lab-2",
+            otherOrganizationName,
+            Topic.FULL_ELR,
+            CustomerStatus.TESTING,
+            "schema1"
+        )
+        settings.receiverStore[receiver2.fullName] = receiver2
+
+        mockkObject(OktaAuthentication.Companion)
+        every { OktaAuthentication.Companion.decodeJwt(any()) } returns
+            TestDefaultJwt(
+                "a.b.c",
+                Instant.now(),
+                Instant.now().plusSeconds(60),
+                claimsMap
+            )
+
+        return SettingsFunction(
+            settingsFacade = facade
+        )
+    }
+
+    private fun setupHttpRequestMessageForTesting(method: HttpMethod = HttpMethod.GET): MockHttpRequestMessage {
+        val httpRequestMessage = MockHttpRequestMessage("{}", method)
+        httpRequestMessage.httpHeaders += mapOf(
+            "authorization" to "Bearer 111.222.333",
+            "content-type" to HttpUtilities.jsonMediaType
+        )
+
+        return httpRequestMessage
+    }
+
+    @Test
+    fun `test access user cannot view full organization settings (getOrganizations)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOrganizations(httpRequestMessage)
```

Comment:
AuthenticatedClaims.authorized call has this for required scopes:

```
*.*.primeadmin
null.*.admin
null.*user
```

and the scope the user has is `test-lab.*.user`

so, given the title of this test, it is not actually testing what it is supposed to?